### DISCUSSION TITLE
Templated item picker

### DISCUSSION THREAD

**Discussion by soreng** [leekelleher/umbraco-contentment#278] [Idea]
Hi,

We are using this package on multiple projects, and love the flexibility it gives us. Especially the "templated list" that allows us to enhance the backoffice with custom experiences like @prjseal showed in https://skrift.io/issues/creating-an-author-picker-using-contentment/

One hurdle of the templated list is when the datasource contains a lot of items.

We would still like the template-feature, but having a picker to select the items.

I have created a POC on a "Templated Item Picker" in a fork, https://github.com/soreng/umbraco-contentment/tree/feature/templated-item-picker, but before opening a PR, I wanted to open an issue to get your views on it @leekelleher  :)

The main changes are the introduction of an implementation of IDataListEditor, mostly copied from `ItemPickerDataListEditor.cs`, and a new view, `template-editor.html` for rendering the list of picked items in the defined template.

**Comment by leekelleher** [comment]
@soreng Quick reply, (as I'm due in a meeting, but wanted to acknowledge the idea).

I definitely like the idea of being able to have a custom UI/view for the Item Picker's "picked" items. (I'm also thinking Cards could be a nice addition). I'll have a think about how it could be implemented into existing Item Picker configuration, as opposed to a separate list-editor.

**Reply by soreng** [reply]
The biggest difference between the ItemPicker and this TempaltedItemPicker is just the addition of a template in the config, and the usage of the template in the view to render the picked items.

The only reason to not just add a template to the item picker, is that the list-editor.html is using the umb-preview-item directive.

**Comment by soreng** [comment]
hi @leekelleher 

I know you are a busy man, but have you had a moment to gather a few thoughts on this subject? :)

**Reply by leekelleher** [reply]
I did prototype something, got the basics working, but I wasn't 100% happy with the markup of the container, (maybe 80% happy 😅).
Another factor is that this would also need to work with the Content Blocks editor - as they all reuse the same "components". But that was pretty close to working too. 😀
